### PR TITLE
[Merged by Bors] - Ultra Fast Super Slick CI

### PR DESCRIPTION
**Pull request by AgeManning**
Attempting to improve our CI speeds as its recently been a pain point.

Major changes:

 - Use a github action to pull stable/nightly rust rather than building it each run
 - Shift test suite to `nexttest` https://github.com/nextest-rs/nextest for CI
 
 UPDATE:

So I've iterated on some changes, and although I think its still not optimal I think this is a good base to start from. Some extra things in this PR:
- Shifted where we pull rust from. We're now using this thing: https://github.com/moonrepo/setup-rust . It's got some interesting cache's built in, but was not seeing the gains that Jimmy managed to get. In either case tho, it can pull rust, cargofmt, clippy, cargo nexttest all in < 5s. So I think it's worthwhile. 
- I've grouped a few of the check-like tests into a single test called `code-test`. Although we were using github runners in parallel which may be faster, it just seems wasteful. There were like 4-5 tests, where we would pull lighthouse, compile it, then run an action, like clippy, cargo-audit or fmt. I've grouped these into a single action, so we only compile lighthouse once, then in each step we run the checks. This avoids compiling lighthouse like 5 times.
- Ive made doppelganger tests run on our local machines to avoid pulling foundry, building and making lcli which are all now baked into the images. 
- We have sccache and do not incremental compile lighthouse

Misc bonus things:
- Cargo update
- Fix web3 signer openssl keys which is required after a cargo update
- Use mock_instant in an LRU cache test to avoid non-deterministic test
- Remove race condition in building web3signer tests

There's still some things we could improve on. Such as downloading the EF tests every run and the web3-signer binary, but I've left these to be out of scope of this PR. I think the above are meaningful improvements.

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -52,139 +54,191 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      if: env.SELF_HOSTED_RUNNERS == false
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          components: rustfmt
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Do we need to install stable rust on hosted runners? they seem to update the image weekly, and is on latest Rust stable version as well

https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md

I also noticed that `moonrepo/setup-rust@v1` caches the `debug` target by default, we could probably make use of it's caching capability (if it's comparable with `rust-cache` but may need `cache-target: release` for some jobs.

Comment (jimmygchen) [reply .github/workflows/test-suite.yml]: oh they've actually done a bit of comparison with `rust-cache` here: https://github.com/moonrepo/setup-rust#swatinemrust-cache

Comment (AgeManning) [reply .github/workflows/test-suite.yml]: Yeah I wasn't sure about updating the stable rust version. 

On actual runs, using this takes < 5s, and in some cases gives us cargo-nextest also. As you've pointed it out, it does misc caching. Figured I'd give it a try. I decided to use debug, as I assumed the debug would cache across runs, but perhaps I should cache release for release builds. 

The old way of getting stable rust is unmaintained, so this seems like the modern way to do it, and avoids misc errors. 

I'll update to cache some release builds and see how it goes.

File: Makefile [hunk 2]
```diff
@@ -108,11 +108,21 @@ build-release-tarballs:
 test-release:
 	cargo test --workspace --release --exclude ef_tests --exclude beacon_chain --exclude slasher
 
+# Runs the full workspace tests in **release**, without downloading any additional
+# test vectors, using nextest.
+nextest-release:
+	cargo nextest run --workspace --release --exclude ef_tests --exclude beacon_chain --exclude slasher
```
Comment (jimmygchen) [inline Makefile:114]: Just reading through the docs in the default nextest config - for CI, it might be useful to run with `--no-fail-fast` and `--failure-output immediate-final`?

https://github.com/sigp/lighthouse/blob/ebd97bfa8390788dde4cd13382eca1fee4411c6a/.config/nextest.toml#L66-L68

https://github.com/sigp/lighthouse/blob/ebd97bfa8390788dde4cd13382eca1fee4411c6a/.config/nextest.toml#L49-L60

File: .github/workflows/test-suite.yml [hunk 3]
```diff
@@ -46,145 +46,196 @@ jobs:
...
-    needs: cargo-fmt
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          components: rustfmt
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: It seems that we don't need `rustfmt` here, so could potentially get rid of this line, but it probably saves nothing (except a line of code) since this step completes in < 5s.

File: Makefile (hunk 2 above)
Comment (michaelsproul) [reply Makefile:114]: I think we might ideally want `fail-fast` for Bors and full testing for PRs. If we had to choose, I'd say fail-fast, because then we get feedback the quickest and free up the runner.

Comment (jimmygchen) [reply Makefile:114]: yeah good point  👍  given the amount of time our CI takes 🐢  (but soon 🐰!! )

Comment (AgeManning) [reply Makefile:114]: Yeah. I saw this too. I thought we wanted to end tests straight up and exit, for resource use. 

Often if there is a bug, usually i can find it without knowing all future test results

File: .github/workflows/test-suite.yml [hunk 4]
```diff
@@ -257,89 +324,65 @@ jobs:
...
     - uses: actions/checkout@v3
-    - name: Install Rust (${{ env.PINNED_NIGHTLY }})
-      run: rustup toolchain install $PINNED_NIGHTLY
-    - name: Install cargo-udeps
-      run: cargo install cargo-udeps --locked --force
+    - name: Get latest version of nightly Rust
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: nightly
+          bins: cargo-udeps
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml:366]: Looking at the logs and looks like it's trying to locate `GITHUB_TOKEN` to use for downloading artifacts:

```
 WARN Failed to read git credential file /home/runner/.git-credentials err=Os { code: 2, kind: NotFound, message: "No such file or directory" }
 WARN Failed to read git credential file /home/runner/.config/git/credentials err=Os { code: 2, kind: NotFound, message: "No such file or directory" }
 WARN Failed to retrieve token from `gh auth token` err=Custom { kind: Other, error: "process exited with `exit status: 1`" }
```

I might be worth supplying the `GITHUB_TOKEN` to prevent getting rate limited by github:

Comment (jimmygchen) [inline .github/workflows/test-suite.yml:366]: Adding `GITHUB_TOKEN` and disable caching here.

Caching is probably not going to be useful for `cargo-udeps` and I think we can turn it off here:

```suggestion
      with:
          channel: nightly
          bins: cargo-udeps
          cache: false
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 5]
```diff
@@ -257,89 +324,65 @@ jobs:
...
-      with:
-        dotnet-version: '6.0.201'
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
+          cache: false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting:

```suggestion
      with:
          channel: stable
          cache-target: release
          components: rustfmt
          cache: false
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 6]
```diff
@@ -46,145 +46,196 @@ jobs:
...
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          components: rustfmt
+          cache-target: release
+          bins: cargo-nextest
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting and remove `rustfmt`:

```suggestion
      with:
          channel: stable
          cache-target: release
          bins: cargo-nextest
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 7]
```diff
@@ -46,145 +46,196 @@ jobs:
...
   release-tests-ubuntu:
     name: release-tests-ubuntu
     # Use self-hosted runners only on the sigp repo.
-    runs-on: ${{ github.repository == 'sigp/lighthouse' && fromJson('["self-hosted", "linux", "large"]') || 'ubuntu-latest'  }}
-    needs: cargo-fmt
+    runs-on: ${{ github.repository == 'sigp/lighthouse' && fromJson('["self-hosted", "linux", "CI", "large"]') || 'ubuntu-latest'  }}
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: We should add single quotes to all of these comparisons, as `${{ }}` evaluates to string, which is why the comparison with `true` doesn't work 

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 8]
```diff
@@ -46,145 +46,196 @@ jobs:
...
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
+          bins: cargo-nextest
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting and remove `rustfmt`:

```suggestion
      with:
          channel: stable
          cache-target: release
          bins: cargo-nextest
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 9]
```diff
@@ -46,145 +46,196 @@ jobs:
...
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          components: rustfmt
+          cache-target: release
+          bins: cargo-nextest
     - name: Install Foundry (anvil)
+      if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 10]
```diff
@@ -46,145 +46,196 @@ jobs:
...
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
+          bins: cargo-nextest
     - name: Install Foundry (anvil)
+      if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 11]
```diff
@@ -231,21 +291,28 @@ jobs:
...
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      if: env.SELF_HOSTED_RUNNERS == false
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
     - name: Install geth
+      if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 12]
```diff
@@ -257,89 +324,65 @@ jobs:
...
+      run:  make arbitrary-fuzz
+    - name: Run cargo audit
+      run: make audit-CI
+    - name: Run cargo vendor to make sure dependencies can be vendored for packaging, reproducibility and archival purpose
+      run:  CARGO_HOME=$(readlink -f $HOME) make vendor
   check-msrv:
     name: check-msrv
     runs-on: ubuntu-latest
-    needs: [cargo-fmt, extract-msrv]
+    needs: [extract-msrv]
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: nitpick: we can probably move the steps from `extract-msrv` to this job as the output isn't used anywhere else. This saves us from one extra git checkout, so probably not super impactful, happy to do this in a separate PR if you prefer.

File: .github/workflows/test-suite.yml [hunk 13]
```diff
@@ -257,89 +324,65 @@ jobs:
...
-    name: clippy
-    runs-on: ubuntu-latest
-    needs: cargo-fmt
-    steps:
-    - uses: actions/checkout@v3
-    - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      env:
+          CARGO_INCREMENTAL: 1
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Is this intended for this step? Not sure if it compiles anything here

File: .github/workflows/test-suite.yml [hunk 14]
```diff
@@ -46,145 +46,196 @@ jobs:
...
+          bins: cargo-nextest
     - name: Install Foundry (anvil)
+      if: env.SELF_HOSTED_RUNNERS == false
       uses: foundry-rs/foundry-toolchain@v1
       with:
         version: nightly-ca67d15f4abd46394b324c50e21e66f306a1162d
     - name: Install make
-      run: choco install -y make
-    - uses: KyleMayes/install-llvm-action@v1
       if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 15]
```diff
@@ -46,145 +46,196 @@ jobs:
...
+      run: choco install -y make
+#    - uses: KyleMayes/install-llvm-action@v1
+#         if: env.SELF_HOSTED_RUNNERS == false
+#      with:
+#        version: "15.0"
+#        directory: ${{ runner.temp }}/llvm
     - name: Set LIBCLANG_PATH
       run: echo "LIBCLANG_PATH=$((gcm clang).source -replace "clang.exe")" >> $env:GITHUB_ENV
+    - name: Install nextest for windows
+      if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 16]
```diff
@@ -46,145 +46,196 @@ jobs:
...
   beacon-chain-tests:
     name: beacon-chain-tests
     # Use self-hosted runners only on the sigp repo.
-    runs-on: ${{ github.repository == 'sigp/lighthouse' && fromJson('["self-hosted", "linux", "large"]') || 'ubuntu-latest'  }}
-    needs: cargo-fmt
+    runs-on: ${{ github.repository == 'sigp/lighthouse' && fromJson('["self-hosted", "linux", "CI", "large"]') || 'ubuntu-latest'  }}
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 17]
```diff
@@ -46,145 +46,196 @@ jobs:
...
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          components: rustfmt
+          bins: cargo-nextest
     - name: Install Foundry (anvil)
+      if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 18]
```diff
@@ -46,145 +46,196 @@ jobs:
...
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]:
```suggestion
```

File: .github/workflows/test-suite.yml [hunk 19]
```diff
@@ -46,145 +46,196 @@ jobs:
...
-    needs: cargo-fmt
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Looks like we don't need Rust for the docker build.

File: .github/workflows/test-suite.yml [hunk 20]
```diff
@@ -46,145 +46,196 @@ jobs:
...
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]:
```suggestion
```

File: .github/workflows/test-suite.yml [hunk 21]
```diff
@@ -194,11 +245,14 @@ jobs:
...
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]:
```suggestion
```

File: .github/workflows/test-suite.yml [hunk 22]
```diff
@@ -208,21 +262,27 @@ jobs:
...
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]:
```suggestion
```

File: .github/workflows/test-suite.yml [hunk 23]
```diff
@@ -208,21 +262,27 @@ jobs:
...
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]:
```suggestion
```

File: .github/workflows/test-suite.yml [hunk 24]
```diff
@@ -231,21 +291,28 @@ jobs:
...
   doppelganger-protection-test:
     name: doppelganger-protection-test
-    runs-on: ubuntu-latest
-    needs: cargo-fmt
+    runs-on: ${{ github.repository == 'sigp/lighthouse' && fromJson('["self-hosted", "linux", "CI", "small"]') || 'ubuntu-latest'  }}
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      if: env.SELF_HOSTED_RUNNERS == false
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: wrapping in single quotes for string comparison:

```suggestion
      if: env.SELF_HOSTED_RUNNERS == 'false'
```

File: .github/workflows/test-suite.yml [hunk 25]
```diff
@@ -231,21 +291,28 @@ jobs:
...
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]:
```suggestion
```

File: .github/workflows/test-suite.yml [hunk 26]
```diff
@@ -231,21 +291,28 @@ jobs:
...
           sudo apt-get update
           sudo apt-get install ethereum
-    - name: Install lighthouse and lcli
+    - name: Install lighthouse
       run: |
           make
-          make install-lcli
+    - name: Install lcli
+      if: env.SELF_HOSTED_RUNNERS == false
+      run: make install-lcli
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml:293]: We lose the check for compatibility between testnet & `lcli` here (that could potentially catch `lcli` breakage), so we might get feedback for this a week late - but I'm sure this has been considered, and I don't mind it if we can get good gains from this.

Comment (jimmygchen) [reply .github/workflows/test-suite.yml:293]: There's also a chance that this test would keep failing  if there's changes that make `lcli` incompatible with the latest BN, and we'd have to update the runner image to the get test to pass.

File: .github/workflows/test-suite.yml [hunk 27]
```diff
@@ -257,89 +324,65 @@ jobs:
...
-    - uses: actions/setup-dotnet@v3
-      with:
-        dotnet-version: '6.0.201'
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Sorry to be annoying here, thought I might as well keep doing this since I've  already started 😅 

```suggestion
```

File: .github/workflows/test-suite.yml [hunk 28]
```diff
@@ -46,145 +46,196 @@ jobs:
...
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
+          bins: cargo-nextest
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting and remove `rustfmt`:

```suggestion
      with:
          channel: stable
          cache-target: release
          bins: cargo-nextest
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 29]
```diff
@@ -46,145 +46,196 @@ jobs:
...
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
+          bins: cargo-nextest
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting and remove `rustfmt`:

```suggestion
      with:
          channel: stable
          cache-target: release
          bins: cargo-nextest
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 30]
```diff
@@ -46,145 +46,196 @@ jobs:
...
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting and remove `rustfmt`:

```suggestion
      with:
          channel: stable
          cache-target: release
          bins: cargo-nextest
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/test-suite.yml [hunk 31]
```diff
@@ -46,145 +46,196 @@ jobs:
...
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
       if: env.SELF_HOSTED_RUNNERS == false
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          components: rustfmt
+          bins: cargo-nextest
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Setting GITHUB_TOKEN to avoid rate limiting and remove `rustfmt`:

```suggestion
      with:
          channel: stable
          bins: cargo-nextest
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by jimmygchen — CHANGES_REQUESTED** [review]
Niiice, we're close! our CI is looking great 🔥 

I've added lots of comments but mostly the same thing (Sorry I thought it would make it easier to batch commit the suggestions, but there are too many occurences!)

So the main things are
- use `GITHUB_TOKEN` for `setup-rust` to avoid getting rate limited (we were getting this quite often before)
- wrap the `env.SELF_HOSTED_RUNNERS` booleans with single quotes as `${{ }}` evaluates to string

Others are just nitpicks, feel free to leave them out ☺️

File: testing/web3signer_tests/tls/lighthouse/cert.pem [hunk 32]
```diff
@@ -1,32 +1,33 @@
 -----BEGIN CERTIFICATE-----
```
Comment (michaelsproul) [inline testing/web3signer_tests/tls/lighthouse/cert.pem:1]: Ah, I also just re-generated these in #4650. You'll get a merge conflict but I don't mind if you just clobber the keys I generated. Explicitly specifying aes256 is probably better than what I did (relying on the default).

File: .github/workflows/test-suite.yml (hunk 13 above)
Comment (AgeManning) [reply .github/workflows/test-suite.yml]: My thinking was that at each step we compile lighthouse and its deps to do the all the various checks. I thought incremental compilation would be beneficial here. This isn't the case? @jimmygchen

File: .github/workflows/test-suite.yml (hunk 26 above)
Comment (AgeManning) [reply .github/workflows/test-suite.yml:293]: Yep, we have considered this. We can add it back, but we figured we'd leave it out until it causes pain. Its not often we have these incompatibilities, i think (famous last words)

Comment (jimmygchen) [reply .github/workflows/test-suite.yml:293]: Sounds good leaving it until it causes pain 👍

File: .github/workflows/test-suite.yml (hunk 13 above)
Comment (jimmygchen) [reply .github/workflows/test-suite.yml]: Oh it makes sense - in that case i think we need to shift this `env` one level up to the _job_ level, e.g. 

```
  check-code:
    name: check-code
    runs-on: ubuntu-latest
    env:
        CARGO_INCREMENTAL: 1
```

At the moment it's scoped at the _step_ level, which i don't think is available to other steps under this job.

File: .github/workflows/test-suite.yml [hunk 33]
```diff
@@ -34,157 +34,195 @@ jobs:
...
+#      with:
+#        version: "15.0"
+#        directory: ${{ runner.temp }}/llvm
     - name: Set LIBCLANG_PATH
       run: echo "LIBCLANG_PATH=$((gcm clang).source -replace "clang.exe")" >> $env:GITHUB_ENV
+    - name: Install nextest for windows
+      if: env.SELF_HOSTED_RUNNERS == 'false'
+      run: |
+        powershell Invoke-WebRequest -Uri https://get.nexte.st/latest/windows -OutFile nextest-windows.zip &&
+        powershell Expand-Archive -Path 'nextest-windows.zip' -DestinationPath 'C:\Users\ContainerAdministrator\.cargo\bin\'
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml]: Do we still need this now the `moonrepo/setup-rust` step installs the `cargo-nextest` binary?

Comment (AgeManning) [reply .github/workflows/test-suite.yml]: Removed this

File: Cargo.lock
Comment (jimmygchen) [inline Cargo.lock:1]: Is deleting this file intentional?

**Review by jimmygchen — APPROVED** [review]
Looks good to me! Can't wait to get this in 🚀

**Comment by AgeManning** [comment]
bors retry

**Comment by paulhauner** [comment]
bors retry

**Comment by paulhauner** [comment]
bors retry
